packages/babel-traverse/src/path/evaluation.js

```diff
@@ -112,25 +112,19 @@ export function evaluate(): { confident: boolean; value: any } {
     }
 
     if (path.isTemplateLiteral()) {
-      let str = "";
-
-      let i = 0;
-      const exprs = path.get("expressions");
-
-      for (const elem of (node.quasis: Array<Object>)) {
-        // not confident, evaluated an expression we don't like
-        if (!confident) break;
+      return evaluateQuasis(path, node.quasis);
+    }
 
-        // add on cooked element
-        str += elem.value.cooked;
+    if (path.isTaggedTemplateExpression()) {
+      const object = path.get("tag.object");
+      const property = path.get("tag.property");
 
-        // add on interpolated expression if it's present
-        const expr = exprs[i++];
-        if (expr) str += String(evaluate(expr));
+      if (
+        object.isIdentifier() && object.node.name === "String" &&
```

Comment:
I think we'll have to check scoping here, since this would also evaluate
```
function fn(String) {
  return String.raw``;
}
fn({raw: (strs, ...args) => ""});
```